sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -789,6 +797,38 @@ case class Range(
     }
   }
 
+  private def computeHistogramStatistics() = {
+    val numBins = conf.histogramNumBins
+    val height = numElements.toDouble / numBins
+    val percentileArray = (0 to numBins).map(i => i * height).toArray
+
+    val binArray = new Array[HistogramBin](numBins)
+    var lowerIndex = percentileArray.head
+    var lowerBinValue = getRangeValue(0)
```

Comment:
It looks we can remove `var` by using `foldLeft` instead of `foreach`. Could you?

Reply:
Yes, updated.